lib/publiccloud/instance.pm

```diff
@@ -91,4 +95,96 @@ sub upload_log {
     assert_script_run("test -d '$tmpdir' && rm -rf '$tmpdir'");
 }
 
+sub wait_for_guestregister
+{
+    my ($self, %args) = @_;
+    $args{timeout} //= 300;
+    my $start_time = time();
+    my $last_info  = 0;
+
+    while (time() - $start_time < $args{timeout}) {
+        my $out = $self->run_ssh_command(cmd => 'sudo systemctl is-active guestregister', proceed_on_failure => 1, quiet => 1);
+        if ($out eq 'inactive') {
+            return time() - $start_time;
+        }
+        if (time() - $last_info > 10) {
+            record_info('WAIT', 'Wait for guest register: ' . $out);
+            $last_info = time();
+        }
+        sleep 1;
+    }
+    die('guestregister didn\'t end in expected timeout=' . $args{timeout});
+}
+
+sub check_ssh_port
+{
+    my ($self, %args) = @_;
+    $args{timeout} //= 600;
+    my $start_time = time();
+
+    while ((my $duration = time() - $start_time) < $args{timeout}) {
+        return $duration if (script_run('nc -vz -w 1 ' . $self->{public_ip} . ' 22', quiet => 1) == 0);
+        sleep 1;
+    }
+    return;
+}
+
+sub get_uptime
+{
+    my ($self) = @_;
+    my $ret    = {days => 0, hours => 0, minutes => 0};
+    my $output = $self->run_ssh_command(cmd => 'uptime');
+    $output = trim($output);
+    my ($uptime) = $output =~ /up\s+(.+),\s+\d+\s+users?/;
+    record_soft_failure("Unable to find uptime - $output") unless (length($output));
+    for my $time (split(/\s*,\s*/, $uptime)) {
+        if ($time =~ /(\d+):(\d+)/) {
+            $ret->{hours}   = $1;
+            $ret->{minutes} = $2;
+        } elsif ($time =~ /(\d+)\s+min/) {
+            $ret->{minutes} = $1;
+        } elsif ($time =~ /(\d+)\s+days?/) {
+            $ret->{days} = $1;
+        } else {
+            record_soft_failure("Unable to parse uptime - $output");
```

Comment:
you know my opinion about such way of using softfailures . or we die here or just record_info

Reply:
The function isn't used, I'm going to remove it.